.github/workflows/build-and-release.yaml

```diff
@@ -0,0 +1,55 @@
+name: "Build and Release"
+
+on: workflow_dispatch
+
+jobs:
+  release:
+    name: Release - ${{ matrix.platform.os-name }}
+    strategy:
+      matrix:
+        platform:
+          - os-name: FreeBSD-x86_64
+            runs-on: ubuntu-20.04
+            target: x86_64-unknown-freebsd
+            skip_tests: true
+
+          - os-name: Linux-x86_64
+            runs-on: ubuntu-20.04
+            target: x86_64-unknown-linux-musl
+
+          - os-name: Linux-aarch64
+            runs-on: ubuntu-20.04
+            target: aarch64-unknown-linux-musl
+
+          - os-name: Linux-riscv64
+            runs-on: ubuntu-20.04
+            target: riscv64gc-unknown-linux-gnu
+
+          - os-name: Windows-x86_64
+            runs-on: windows-latest
+            target: x86_64-pc-windows-msvc
+
+          - os-name: macOS-x86_64
+            runs-on: macOS-latest
+            target: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.platform.runs-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Build binary
+        uses: houseabsolute/actions-rust-cross@v0
+        with:
+          working-directory: echidna-lb
+          command: ${{ matrix.platform.command }}
+          target: ${{ matrix.platform.target }}
+          args: "--locked --release"
```

Comment:
Should this binary be statically linked?

Reply:
Yes, better not to keep an OS dependency. Can you please suggest how to add it?

Reply:
Maybe add the arch as a matrix field, and try to build using musl?
```yaml
          - os-name: Linux-x86_64
            runs-on: ubuntu-20.04
            target: x86_64-unknown-linux-musl
            arch: x86_64

          - os-name: Linux-aarch64
            runs-on: ubuntu-20.04
            target: aarch64-unknown-linux-musl
            arch: aarch64
```

```yaml
          args: "--locked --release --target=${{ matrix.arch }}-unknown-linux-musl
```
Mac should work the same way I think.
You'd have to take a stab at riscv64 on your own..

Reply:
The action uses **cross** for cross-compilation. Will this arg work with `cross build` command? Ref: https://github.com/houseabsolute/actions-rust-cross

Reply:
Hmmm... I don't know. Let's maybe do the static linking bit separately in a different PR..